test.sh

```diff
@@ -42,35 +42,53 @@ test_run() {
     TARGET=$2
     FLAVOR=$3
 
+    local TARGET_MAJOR=${PGTARGET%%.*}
+    if [ "$TARGET_MAJOR" -eq 18 ]; then
+        PGAUTO_COMPOSE="docker-compose-pgauto18.yml"
+        PGTARGET_COMPOSE="docker-compose-pgtarget18.yml"
+        PGVERSION_FILE="postgres-data/18/docker/PG_VERSION"
+    else
+        PGAUTO_COMPOSE="docker-compose-pgauto.yml"
+        PGTARGET_COMPOSE="docker-compose-pgtarget.yml"
+        PGVERSION_FILE="postgres-data/PG_VERSION"
+    fi
+
     # Delete any existing test PostgreSQL data
     if [ -d postgres-data ]; then
         echo "Removing old PostgreSQL data from test directory"
         sudo rm -rf postgres-data
     fi
 
+    export TARGET_TAG="${TARGET}-${FLAVOR}"
+
     # Start an empty pgautoupgrade container to make sure this is possible as well
-    TARGET_TAG="${TARGET}-${FLAVOR}" docker compose -f docker-compose-pgauto.yml up --wait -d
+    docker compose -f $PGAUTO_COMPOSE up --wait -d
 
     # Shut down any containers that are still running
-    docker compose -f docker-compose-pgauto.yml down --remove-orphans
+    docker compose -f $PGAUTO_COMPOSE down --remove-orphans
 
     # Delete the upgraded PostgreSQL data directory
     sudo rm -rf postgres-data
 
     import_adventure_works
 
     # Start pgautoupgrade container
-    TARGET_TAG="${TARGET}-${FLAVOR}" docker compose -f docker-compose-pgauto.yml up --wait -d
+    docker compose -f $PGAUTO_COMPOSE up --wait -d
 
     # Verify the PostgreSQL data files are now the target version
-    PGVER=$(sudo cat postgres-data/PG_VERSION)
+    PGVER=$(sudo cat $PGVERSION_FILE)
     if [ "$PGVER" != "${TARGET%%.*}" ]; then
         banner '*' "Standard automatic upgrade of PostgreSQL from version ${VERSION} to ${TARGET} FAILED!"
         FAILURE=1
     else
         banner '*' "Standard automatic upgrade of PostgreSQL from version ${VERSION} to ${TARGET} SUCCEEDED!"
     fi
 
+    if ! docker compose -f $PGAUTO_COMPOSE exec postgres psql -U postgres -tAc "SELECT 1 FROM pg_database WHERE datname='AdventureWorks'" | grep -q 1; then
```

Comment:
Oh, are we bundling the Adventure works database?

Reply:
nono, but it is the database used during testing to verify the upgrade process.

Reply:
Ahhh.  I somehow missed this was being run in `test.sh` rather than a check in the standard upgrade process.  No worries. :smile: